app/controlplane/plugins/core/slack-webhook/v1/slack_webhook.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/go-kratos/kratos/v2/log"
+)
+
+type Integration struct {
+	*sdk.FanOutIntegration
+}
+
+// 1 - API schema definitions
+type registrationRequest struct {
+	WebhookURL string `json:"webhook" jsonschema:"format=uri,description=URL of the slack webhook"`
+}
+
+type attachmentRequest struct{}
+
+// 2 - Configuration state
+type registrationState struct{}
+
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanOut(
+		&sdk.NewParams{
+			ID:          "slack-webhook",
+			Version:     "1.0",
+			Description: "Send attestations to Slack",
+			Logger:      l,
+			InputSchema: &sdk.InputSchema{
+				Registration: registrationRequest{},
+				Attachment:   attachmentRequest{},
+			},
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &Integration{base}, nil
+}
+
+// Register is executed when a operator wants to register a specific instance of this integration with their Chainloop organization
+func (i *Integration) Register(_ context.Context, req *sdk.RegistrationRequest) (*sdk.RegistrationResponse, error) {
+	i.Logger.Info("registration requested")
+
+	var request *registrationRequest
+	if err := sdk.FromConfig(req.Payload, &request); err != nil {
+		return nil, fmt.Errorf("invalid registration request: %w", err)
+	}
+
+	if err := executeWebhook(request.WebhookURL, "This is a test message. Welcome to Chainloop!"); err != nil {
+		return nil, fmt.Errorf("error validating a webhook: %w", err)
+	}
+
+	// Configuration State
+	config, err := sdk.ToConfig(&registrationState{})
+	if err != nil {
+		return nil, fmt.Errorf("marshalling configuration: %w", err)
+	}
+
+	return &sdk.RegistrationResponse{
+		Configuration: config,
+		// We treat the webhook URL as a sensitive field so we store it in the credentials storage
+		Credentials: &sdk.Credentials{Password: request.WebhookURL},
+	}, nil
+}
+
+// Attachment is executed when to attach a registered instance of this integration to a specific workflow
+func (i *Integration) Attach(_ context.Context, _ *sdk.AttachmentRequest) (*sdk.AttachmentResponse, error) {
+	i.Logger.Info("attachment requested")
+	return &sdk.AttachmentResponse{}, nil
+}
+
+// Execute will be instantiate when either an attestation or a material has been received
+// It's up to the plugin builder to differentiate between inputs
+func (i *Integration) Execute(_ context.Context, req *sdk.ExecutionRequest) error {
+	i.Logger.Info("execution requested")
+
+	if err := validateExecuteRequest(req); err != nil {
+		return fmt.Errorf("running validation: %w", err)
+	}
+
+	var config *registrationState
+	if err := sdk.FromConfig(req.RegistrationInfo.Configuration, &config); err != nil {
+		return fmt.Errorf("invalid registration config: %w", err)
+	}
+
+	attestationJSON, err := json.MarshalIndent(req.Input.Attestation.Statement, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error marshaling JSON: %w", err)
+	}
+
+	metadata := req.ChainloopMetadata
+	a := fmt.Sprintf("\n```\n%s\n```\n", string(attestationJSON))
```

Comment:
Can't this be part of the template itself?

Reply:
I was not able to make those backticks part of the template ...

Reply:
Interesting, I'd then add a comment and we can move on.

Reply:
It seems it's possible only with `print` function. TBH I think you approach is less obscure. I'd just add a comment indicating why we add the backticks outside.

https://go.dev/play/p/uzFGNTZEGvj